pkg/config/expand.go

```diff
@@ -186,6 +187,33 @@ func (bp Blueprint) expandBackend(grp *Group) {
 	}
 }
 
+func kubectlProviderRequiredModule(grp *Group) (bool, Module) {
+	for _, mod := range grp.Modules {
+		if strings.Contains(mod.Source, "gke-cluster") || strings.Contains(mod.Source, "pre-existing-gke-cluster") {
+			return true, mod
+		}
+	}
+	return false, Module{}
+}
+
+func getModuleKubectlProviders(mod Module) map[string]TerraformProvider {
+	kubectlConf := Dict{}
+	for s, v := range map[string]string{
+		"cluster_ca_certificate": "cluster_ca_certificate",
+		"host":                   "host_endpoint",
+		"token":                  "access_token"} {
+		kubectlConf = kubectlConf.With(s, ModuleRef(mod.ID, v).AsValue())
+	}
+	kubectlConf = kubectlConf.
+		With("apply_retry_count", cty.NumberIntVal(15)).
+		With("load_config_file", cty.BoolVal(false))
```

Comment:
Can we avoid hardcoded `load_config_file`, `apply_retry_count` and read it from the internal provider if they exist there?

Reply:
These provider config are created in root module on gcluster create/deploy ops, so there seems to be no internal provider which can be referred.